### ISSUE TITLE
[Research] Add PHQ-9/GAD-7 Mental Health Questionnaires for Supervised Training Data

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#25]
## What

Add **PHQ-9 (depression)** and **GAD-7 (anxiety)** standardized questionnaires to the study profile page. This creates labeled ground-truth data for training supervised mental health prediction models from EEG + Apple Health features.

## Why

- Current system has **no ground-truth mental health labels** — all emotion predictions are unlabeled
- PHQ-9 and GAD-7 are the gold-standard clinical screening tools (used in 90%+ of digital mental health studies)
- The SWARTS-DA protocol (BMJ Open 2025) validates this exact approach: combine active self-reports (PHQ-9/GAD-7) with passive wearable data for depression/anxiety screening
- With labeled data, we can train supervised models instead of relying on heuristics
- Self-report labels also serve as validation for EEG emotion predictions

### Step 1: Add questionnaire data model
```typescript
// shared/schema.ts — add new table
export const mentalHealthAssessments = pgTable("mental_health_assessments", {
  id: serial("id").primaryKey(),
  userId: integer("user_id").references(() => users.id),
  assessmentType: text("assessment_type"), // "PHQ-9" | "GAD-7"
  score: integer("score"),
  severity: text("severity"), // "minimal" | "mild" | "moderate" | "moderately_severe" | "severe"
  responses: jsonb("responses"), // individual item responses
  eegSessionId: integer("eeg_session_id"), // link to nearest EEG session
  createdAt: timestamp("created_at").defaultNow(),
});
```

### Step 2: PHQ-9 implementation (9 items, 0-3 scale each)
Questions cover: interest/pleasure, feeling down, sleep, energy, appetite, self-worth, concentration, psychomotor, suicidal ideation.

Scoring: 0-4 minimal, 5-9 mild, 10-14 moderate, 15-19 moderately severe, 20-27 severe.

### Step 3: GAD-7 implementation (7 items, 0-3 scale each)
Questions cover: nervousness, uncontrollable worry, excessive worry, trouble relaxing, restlessness, irritability, feeling afraid.

Scoring: 0-4 minimal, 5-9 mild, 10-14 moderate, 15-21 severe.

### Step 4: Study mode integration
- Prompt user to complete PHQ-9 + GAD-7 weekly (standard clinical interval)
- Link each assessment to the nearest EEG session for paired analysis
- Show trend chart of scores over time on study profile page
- Add notification/reminder for weekly assessment

### Step 5: ML training data pipeline
- Export paired data: (eeg_features, health_features, phq9_score, gad7_score)
- Train regression model: EEG + Apple Health → PHQ-9/GAD-7 score prediction
- Validate: predicted vs actual score correlation

### Files to Modify
- `shared/schema.ts` — Add mental_health_assessments table
- `server/routes.ts` — Add CRUD endpoints for assessments
- `client/src/pages/study-profile.tsx` — Add questionnaire UI
- `client/src/components/phq9-form.tsx` (NEW) — PHQ-9 questionnaire component
- `client/src/components/gad7-form.tsx` (NEW) — GAD-7 questionnaire component
- `ml/training/train_mental_health.py` (NEW) — Supervised model training

### Expected Impact
- Creates labeled dataset for supervised mental health prediction
- Enables validation of EEG emotion predictions against clinical instruments
- Over 4-8 weeks of data: can predict PHQ-9/GAD-7 from EEG + Apple Health

## Complexity
LOW-MEDIUM — standard questionnaire UI + database table + API endpoints

## References
- SWARTS-DA protocol (BMJ Open 2025): https://pmc.ncbi.nlm.nih.gov/articles/PMC12182151/
- PHQ-9: Kroenke et al. (2001), Journal of General Internal Medicine
- GAD-7: Spitzer et al. (2006), Archives of Internal Medicine
- Digital phenotyping for depression (JMIR 2025): https://mental.jmir.org/2025/1/e63622
